core/src/nucleus/ribosome/callback/validation_package.rs

```diff
@@ -0,0 +1,57 @@
+extern crate serde_json;
+use context::Context;
+use holochain_core_types::{entry_type::EntryType, error::HolochainError};
+use nucleus::{
+    ribosome::{
+        self,
+        callback::{get_dna, get_wasm, CallbackResult},
+    },
+    ZomeFnCall,
+};
+use std::sync::Arc;
+
+pub fn get_validation_package_definition(
+    entry_type: EntryType,
+    context: Arc<Context>,
+) -> Result<CallbackResult, HolochainError> {
+    match entry_type {
+        EntryType::App(app_entry_type) => {
+            let dna = get_dna(&context).expect("Callback called without DNA set!");
+            let zome_name = dna.get_zome_name_for_entry_type(&app_entry_type);
+            if zome_name.is_none() {
+                return Ok(CallbackResult::NotImplemented);
+            }
+
+            let zome_name = zome_name.unwrap();
+            match get_wasm(&context, &zome_name) {
+                None => Err(HolochainError::ErrorGeneric(String::from("no wasm found"))),
```

Comment:
How do you feel about the cyclomatic complexity here and the level of  matches? Could we make use of error map functions?

e.g.  

let wasm = get_wasm(context,&zomename).ok_or(<Error goes here>)?;
let result = run_dna(...).map_err(<Error goes here>)?